qutip/cy/dicke.pyx

```diff
@@ -0,0 +1,645 @@
+
+        # perform loop in each row of matrix
+        for r in jmm1_row:
+            j, m, m1 = jmm1_row[r]
+            jmm1_1 = (j, m, m1)
+            jmm1_2 = (j, m + 1, m1 + 1)
+            jmm1_3 = (j + 1, m + 1, m1 + 1)
+            jmm1_4 = (j - 1, m + 1, m1 + 1)
+            jmm1_5 = (j + 1, m, m1)
+            jmm1_6 = (j - 1, m, m1)
+            jmm1_7 = (j + 1, m - 1, m1 - 1)
+            jmm1_8 = (j, m - 1, m1 - 1)
+            jmm1_9 = (j - 1, m - 1, m1 - 1)
+
+            g1 = self.gamma1(jmm1_1)
+            c1 = jmm1_inv[jmm1_1]
+
+            lindblad_row.append(int(r))
+            lindblad_col.append(int(c1))
+            lindblad_data.append(g1)
+
+            # generate gammas in the given row
+            # check if the gammas exist
+            # load gammas in the lindbladian in the correct position
+
+            if jmm1_2 in jmm1_inv:
+                g2 = self.gamma2(jmm1_2)
+                c2 = jmm1_inv[jmm1_2]
+
+                lindblad_row.append(int(r))
+                lindblad_col.append(int(c2))
+                lindblad_data.append(g2)
+
+            if jmm1_3 in jmm1_inv:
+                g3 = self.gamma3(jmm1_3)
+                c3 = jmm1_inv[jmm1_3]
+
+                lindblad_row.append(int(r))
+                lindblad_col.append(int(c3))
+                lindblad_data.append(g3)
+
+            if jmm1_4 in jmm1_inv:
+                g4 = self.gamma4(jmm1_4)
+                c4 = jmm1_inv[jmm1_4]
+
+                lindblad_row.append(int(r))
+                lindblad_col.append(int(c4))
+                lindblad_data.append(g4)
+
+            if jmm1_5 in jmm1_inv:
+                g5 = self.gamma5(jmm1_5)
+                c5 = jmm1_inv[jmm1_5]
+
+                lindblad_row.append(int(r))
+                lindblad_col.append(int(c5))
+                lindblad_data.append(g5)
+
+            if jmm1_6 in jmm1_inv:
+                g6 = self.gamma6(jmm1_6)
+                c6 = jmm1_inv[jmm1_6]
+
+                lindblad_row.append(int(r))
+                lindblad_col.append(int(c6))
+                lindblad_data.append(g6)
+
+            if jmm1_7 in jmm1_inv:
+                g7 = self.gamma7(jmm1_7)
+                c7 = jmm1_inv[jmm1_7]
+
+                lindblad_row.append(int(r))
+                lindblad_col.append(int(c7))
+                lindblad_data.append(g7)
+
+            if jmm1_8 in jmm1_inv:
+                g8 = self.gamma8(jmm1_8)
+                c8 = jmm1_inv[jmm1_8]
+
+                lindblad_row.append(int(r))
+                lindblad_col.append(int(c8))
+                lindblad_data.append(g8)
+
+            if jmm1_9 in jmm1_inv:
+                g9 = self.gamma9(jmm1_9)
+                c9 = jmm1_inv[jmm1_9]
+
+                lindblad_row.append(int(r))
+                lindblad_col.append(int(c9))
+                lindblad_data.append(g9)
+
+        cdef lindblad_matrix = csr_matrix((lindblad_data, (lindblad_row, lindblad_col)),
```

Comment:
pep8: line character overrun here (max 79 chars per line)
A few other examples in this file too